src/raftstore/store/worker/read.rs

```diff
@@ -0,0 +1,895 @@
+        self.region = Some(region);
+    }
+
+    pub fn set_term(&mut self, term: u64) {
+        self.term = Some(term);
+    }
+
+    pub fn set_applied_index_term(&mut self, applied_index_term: u64) {
+        self.applied_index_term = Some(applied_index_term);
+    }
+
+    pub fn set_leader_lease(&mut self, lease: RemoteLease) {
+        self.leader_lease = Some(lease);
+    }
+}
+
+pub enum Task {
+    Register(ReadDelegate),
+    Update((u64, Progress)),
+    Read(StoreMsg),
+    Destroy(u64),
+}
+
+impl Task {
+    pub fn register(peer: &Peer) -> Task {
+        let delegate = ReadDelegate::from_peer(peer);
+        Task::Register(delegate)
+    }
+
+    pub fn update(region_id: u64, progress: Progress) -> Task {
+        Task::Update((region_id, progress))
+    }
+
+    pub fn destroy(region_id: u64) -> Task {
+        Task::Destroy(region_id)
+    }
+
+    #[inline]
+    pub fn acceptable(msg: &StoreMsg) -> bool {
+        match *msg {
+            StoreMsg::RaftCmd { ref request, .. } => {
+                if request.has_admin_request() || request.has_status_request() {
+                    false
+                } else {
+                    for r in request.get_requests() {
+                        match r.get_cmd_type() {
+                            CmdType::Get | CmdType::Snap => (),
+                            CmdType::Delete
+                            | CmdType::Put
+                            | CmdType::DeleteRange
+                            | CmdType::Prewrite
+                            | CmdType::IngestSST
+                            | CmdType::Invalid => return false,
+                        }
+                    }
+                    true
+                }
+            }
+            StoreMsg::BatchRaftSnapCmds { .. } => true,
+            _ => false,
+        }
+    }
+}
+
+impl Display for Task {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match *self {
+            Task::Register(ref delegate) => write!(f, "localreader Task::Register {:?}", delegate),
+            Task::Read(ref msg) => write!(f, "localreader Task::Msg {:?}", msg),
+            Task::Update(ref progress) => write!(f, "localreader Task::Update {:?}", progress),
+            Task::Destroy(region_id) => write!(f, "localreader Task::Destroy region {}", region_id),
+        }
+    }
+}
+
+fn server_busy(cmd: StoreMsg) {
+    let mut err = errorpb::Error::new();
+    err.set_message(RAFTSTORE_IS_BUSY.to_owned());
+    let mut server_is_busy = errorpb::ServerIsBusy::new();
+    server_is_busy.set_reason(RAFTSTORE_IS_BUSY.to_owned());
+    err.set_server_is_busy(server_is_busy);
+    let mut resp = RaftCmdResponse::new();
+    resp.mut_header().set_error(err);
+
+    let read_resp = ReadResponse {
+        response: resp,
+        snapshot: None,
+    };
+
+    match cmd {
+        StoreMsg::RaftCmd { callback, .. } => callback.invoke_read(read_resp),
+        StoreMsg::BatchRaftSnapCmds {
```

Comment:
Do we need to remove batch snapshot after this pr merged? In most product cases, we find that the effect  of batch snapshot is not very well, and it also make the code a little complicate.